manager/controlapi/service.go

```diff
@@ -175,6 +176,45 @@ func validateServiceSpec(spec *api.ServiceSpec) error {
 	return nil
 }
 
+func (s *Server) checkPortConflicts(spec *api.ServiceSpec) error {
+	if spec.Endpoint == nil {
+		return nil
+	}
+
+	var (
+		services []*api.Service
+		err      error
+	)
+
+	s.store.View(func(tx store.ReadTx) {
+		services, err = store.FindServices(tx, store.All)
+	})
+	if err != nil {
+		return err
+	}
+
+	pcToString := func(pc *api.PortConfig) string {
+		port := strconv.FormatUint(uint64(pc.PublishedPort), 10)
+		return port + "/" + pc.Protocol.String()
+	}
+
+	usedPorts := make(map[string]bool)
+	for _, service := range services {
```

Comment:
May need to check service.Endpoint for allocated ports, as well.

This may have been the original reason why we didn't do this.